### ISSUE TITLE
Font Size not as flexible as before

### ISSUE THREAD

**Issue by Icelain** [neovide/neovide#1185]
**Describe the bug**
Updated to the latest neovide-git through the aur and fonts are not as flexible as they were before. For example, earlier(on a 2-3 month old build) I could set the font size to be 12.3 or something similar and it would adjust just fine. Now even setting the font size to 12 from 13 shows no difference.

**To Reproduce**
1. Set the font to jetbrains mono .
2. Compare changes between font size 12 and 13.

**Expected behavior**
Neovide should show fonts with decimal sizes like before and there should be a difference in the sizes 12 and 13. I know I'm being a bit nitpicky but otherwise the font is either too large or too small.

**Screenshots**
Eg(the font is set to 12 and 13 in the different instances)- ![img](https://i.imgur.com/9QQEo9X.png)

**Desktop (please complete the following information):**
 - OS: Arch Linux XFCE4
 - Neovide Version 0.8.0
 - Neovim Version 0.6.1

**Please run `neovide --log` and paste the contents of the `.log` file here:**

   TRACE [neovide] Neovide version: 0.8.0

**Comment by fredizzimo** [comment]
I will check this. I agree that there should be a difference, and I can repeat it here. I know the reason, Neovide scales the font so that the width is a an integer multiplier, which seems to match what almost all other editors do, in my test. Still there is some minor difference, because I tested with notepad in Windows, and there there's a difference there between the two sizes. So I will try to figure out the exact correct calculation.

Fractional font sizes in pt can easily be added, and that would probably make it possible to workaround the issue. The actual width would still be rounded, but it would give more control, to not skip two pixels.

**Comment by fredizzimo** [comment]
I will look at it a bit more in the evening, but I don't think there's much more that can be done actually, without supporting subpixel positioning, which is very tricky since we need to be grid based. As you can see exact pixel widths are represented, so allowing input of fractional sizes would not actually help. 

TRACE [neovide::renderer::fonts::caching_shaper] Updating font: JetBrainsMono NF:h11
TRACE [neovide::renderer::fonts::caching_shaper] Font updated to: JetBrainsMono NF:h11
TRACE [neovide::renderer::fonts::caching_shaper] Using font_size: 14.67px
TRACE [neovide::renderer::fonts::caching_shaper] Font width: 8.80px 8.80px
TRACE [neovide::renderer::fonts::caching_shaper] Fudge factor: 1.02
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font size: 15.00px
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font width: 9.00px
TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 9, height: 20 }

Updating font: JetBrainsMono NF:h12
TRACE [neovide::renderer::fonts::caching_shaper] Font updated to: JetBrainsMono NF:h12
TRACE [neovide::renderer::fonts::caching_shaper] Using font_size: 16.00px
TRACE [neovide::renderer::fonts::caching_shaper] Font width: 9.60px 9.60px
TRACE [neovide::renderer::fonts::caching_shaper] Fudge factor: 1.04
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font size: 16.67px
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font width: 10.00px
TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 10, height: 22 }

TRACE [neovide::renderer::fonts::caching_shaper] Updating font: JetBrainsMono NF:h13
TRACE [neovide::renderer::fonts::caching_shaper] Font updated to: JetBrainsMono NF:h13
TRACE [neovide::renderer::fonts::caching_shaper] Using font_size: 17.33px
TRACE [neovide::renderer::fonts::caching_shaper] Font width: 10.40px 10.40px
TRACE [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.96
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font size: 16.67px
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font width: 10.00px
TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 10, height: 22 }

TRACE [neovide::renderer::fonts::caching_shaper] Updating font: JetBrainsMono NF:h14
TRACE [neovide::renderer::fonts::caching_shaper] Font updated to: JetBrainsMono NF:h14
TRACE [neovide::renderer::fonts::caching_shaper] Using font_size: 18.67px
TRACE [neovide::renderer::fonts::caching_shaper] Font width: 11.20px 11.20px
TRACE [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.98
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font size: 18.33px
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font width: 11.00px
TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 11, height: 25 }

Notepad seems to use fractional sizes based on these images. 

![image](https://user-images.githubusercontent.com/9946255/151388040-42be12c3-9981-4a68-a7c5-730764655493.png)

But Windows terminal renders the exact same size for 12 and 13, just like we do
![image](https://user-images.githubusercontent.com/9946255/151392253-2af22e29-30b8-40de-8bd1-ca2dd3b82dda.png)

And Neovim has the exact same sizes
![image](https://user-images.githubusercontent.com/9946255/151393924-322b2f8c-ddcc-41c6-9886-1ba4a8e72918.png)

I checked the sizes by zooming in and measuring the exact amount of pixels.

**Comment by last-partizan** [comment]
@Icelain try using fractional size: line `12.0` and `13.0` it should work as before.

**Comment by fredizzimo** [comment]
Note that, it will change the proportions of the fonts slightly though, and at least to my eyes look quite cramped. It's not that bad for Jetbrains, since the nominal size is very close to full pixels, For example size 14 should be 14.20 pixels wide, but will be rendered in a 14 pixel box. For fonts and sizes close to midway between pixels it will look a bit worse. The thickness might also vary, due to antialiasing effects.  

The not fudged version is how it looks with 14.0pt, the bigger linespacing in the webbrowser versions are due to css, so ignore that. The browsers don't have any subpixel positioning, while notepad at the bottom has. Notice that it's height is the same as the non-fudged version, but the spacing is correct, so it looks typographically right. (click the image to view it at the full resolution)
![image](https://user-images.githubusercontent.com/9946255/151436422-619358be-b0dc-4442-ad5f-d287f15870d0.png)

**Comment by fredizzimo** [comment]
I had to quickly measure, since I looked at the image again, and the notepad line looked to short to explain the 4% difference there should be in the width if the font was rendered exactly according to the size. It seems like it uses some other approximation or rounding to make the font appear slightly larger in height. So I compared it directly to the Non-fudged version, which has the same height, and actually the same width as well.
![image](https://user-images.githubusercontent.com/9946255/151445603-10be2c8b-4d9d-4509-b42d-44b9ecf5c120.png)

So it's actually using the same rounded grid size, but the letters are better positioned or scaled (aliased) within that grid. I don't have time to analyze it in detail, but they might actually render it with a scaled width. 

Anyways, the difference is quite small, although bigger with a font size of 8, where the width is 0.4 pixels oversized. If you click on the image to open it in full size, then at least for me the left and the notepad one is considerably easier to read.

![image](https://user-images.githubusercontent.com/9946255/151453683-e5f9fbc1-5720-4e04-8a7d-2e69dbebf0fe.png)
